src/Driver/PDO/Connection.php

```diff
@@ -87,17 +91,80 @@ public function quote(string $value): string
         return $this->connection->quote($value);
     }
 
-    public function lastInsertId(?string $name = null): string
+    /**
+     * {@inheritDoc}
+     */
+    public function lastInsertId()
     {
+        $driverName = $this->connection->getAttribute(PDO::ATTR_DRIVER_NAME);
+
+        if ($driverName === 'oci') {
+            throw IdentityColumnsNotSupported::new();
+        }
```

Comment:
Just as demonstration of an alternative to subclassing. Please see [this comment](https://github.com/doctrine/dbal/pull/4324/files#r499686504).

Reply:
This would work too. I don't think that sub-classing in this case is much better than if-statements. If it doesn't affect the API, we can change it as soon as needed.

Reply:
Unless I didn't get you, this contradicts https://github.com/doctrine/dbal/pull/4324#discussion_r500568276. Which way do you want to go?

Reply:
The linked comment clarifies that I was suggesting an individual class per driver, not an abstract decorator. Although I originally suggested to use driver-specific classes, I'm fine if we start without them.

Reply:
Actually, let's start without decorators. A few extra conditions in `PDO\Connection` don't look too bad, and add much less clutter than creating a decorator for each platform. Please let me know if you're still happy with that.